### PR TITLE
Clean up tests, add a redispatch unit test

### PR DESCRIPTION
**Pull request by incompl**

### CODE REVIEW

File: scripts/server.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ const nameify = (str) =>
 const srces = {
   css: 'src/plugin.scss',
   js: 'src/plugin.js',
-  tests: glob.sync('test/**/*.test.js')
+  tests: glob.sync('test/**/test.*.js')
```
Comment (incompl) [inline scripts/server.js]: This is to have shorter, easier-to-read file names that still all start with the same so they sort together

File: test/ads.html [hunk 2]
```diff
@@ -1,33 +0,0 @@
-<!DOCTYPE html>
```
Comment (incompl) [inline test/ads.html]: This file was way old and broken. I don't think it's worth fixing / maintaining

File: test/events.backup.js [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import '../example/app.js';
```
Comment (incompl) [inline test/events.backup.js]: I didn't really fix this file, but I think this is one thing it'd need. The jury is out on whether we fix or rewrite this file.

File: test/events.html [hunk 4]
```diff
@@ -1,47 +0,0 @@
-<!DOCTYPE html>
```
Comment (incompl) [inline test/events.html]: Also old, broken, and I don't think worth fixing

File: test/test.redispatch.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    // Redispatch helper for tests
+    this.redispatch = function(type) {
+      const event = {type};
+
+      eventBuffer = [];
+      redispatch.call(this.player, event);
+
+      if (eventBuffer.length === 1) {
+        return eventBuffer[0].type;
+      } else if (event.cancelBubble) {
```
Comment (ldayananda) [inline test/test.redispatch.js]: Is this possible with how event wraps type?

File: test/test.redispatch.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+      ads: {
+        state: 'content-set',
+
+        snapshot: {
+          ended: false,
+          currentSrc: 'my vid'
+        },
+
+        videoElementRecycled() {
```
Comment (ldayananda) [inline test/test.redispatch.js]: would be really, really nice to have another test for the `return true` case at the same time.

File: test/test.redispatch.js (hunk 5 above)
Comment (incompl) [inline test/test.redispatch.js]: Redispatch can add cancelBubble to the mocked event, even though it's not a real event.

I added a new unit test to this PR to show that it works.

File: test/test.redispatch.js (hunk 6 above)
Comment (incompl) [inline test/test.redispatch.js]: Added a test

**Comment by ldayananda** [comment]
LGTM

**Pull request merged**